v2/fake/fake.go

```diff
@@ -249,62 +249,217 @@ func UnexpectedActionError() error {
 	return errors.New("Unexpected action")
 }
 
-// CatalogReaction is sent as the response to GetCatalog requests.
+// CatalogReactionInterface is sent as the response to GetCatalog requests.
+type CatalogReactionInterface interface {
+	react() (*v2.CatalogResponse, error)
+}
+
 type CatalogReaction struct {
 	Response *v2.CatalogResponse
 	Error    error
 }
 
-// ProvisionReaction is sent as the response ProvisionInstance requests.
+func (r *CatalogReaction) react() (*v2.CatalogResponse, error) {
+	if r == nil {
+		return nil, UnexpectedActionError()
+	}
+	return r.Response, r.Error
+}
+
+type DynamicCatalogReaction func() (*v2.CatalogResponse, error)
+
+func (r DynamicCatalogReaction) react() (*v2.CatalogResponse, error) {
+	return r()
+}
+
+// ProvisionReactionInterface is sent as the response ProvisionInstance requests.
+type ProvisionReactionInterface interface {
+	react(*v2.ProvisionRequest) (*v2.ProvisionResponse, error)
+}
+
 type ProvisionReaction struct {
 	Response *v2.ProvisionResponse
 	Error    error
 }
 
-// UpdateInstanceReaction is sent as the response UpdateInstance requests.
+func (r *ProvisionReaction) react(_ *v2.ProvisionRequest) (*v2.ProvisionResponse, error) {
+	if r == nil {
+		return nil, UnexpectedActionError()
+	}
+	return r.Response, r.Error
+}
+
+type DynamicProvisionReaction func(*v2.ProvisionRequest) (*v2.ProvisionResponse, error)
+
+func (r DynamicProvisionReaction) react(req *v2.ProvisionRequest) (*v2.ProvisionResponse, error) {
+	return r(req)
+}
+
+// UpdateInstanceReactionInterface is sent as the response UpdateInstance requests.
```

Comment:
About half of these comments are missing the "to" in "the response to X requests"

Reply:
lol. I changed the wrong comments.